src/deploy/functions/backend.ts

```diff
@@ -0,0 +1,356 @@
+import * as proto from "../../gcp/proto";
+import * as gcf from "../../gcp/cloudfunctions";
+import * as cloudscheduler from "../../gcp/cloudscheduler";
+import * as utils from "../../utils";
+import { FirebaseError } from "../../error";
+import { Context } from "./args";
+
+export interface ScheduleRetryConfig {
+  retryCount?: number;
+  maxRetryDuration?: proto.Duration;
+  minBackoffDuration?: proto.Duration;
+  maxBackoffDuration?: proto.Duration;
+  maxDoublings?: number;
+}
+
+export interface PubSubSpec {
+  id: string;
+  project: string;
+
+  // What we're actually planning to invoke with this topic
+  targetService: FunctionSpec;
+}
+
+export interface ScheduleSpec {
+  id: string;
+  project: string;
+  // Note: schedule is missing in the existingBackend because we
+  // don't actually spend the API call looking up the schedule;
+  // we just infer it from function labels.
```

Comment:
```suggestion
  // we just infer identifiers from function labels.
```

Tiny nit, but when I first read this I thought "it" meant the actual schedule itself.